### DISCUSSION TITLE
Disable JsonIdentityInfo serialization

### DISCUSSION THREAD

**Discussion by jwgmeligmeyling** [FasterXML/jackson-databind#4619] [Ideas]
Some clients may not support `@JsonIdentityInfo`. It would then be convenient to disable this feature on a `ObjectMapper` level.

I've found this question asked before on Stack Overflow: https://stackoverflow.com/questions/29991236/ignore-the-jackson-jsonidentityinfo-during-serialization . The suggested solution there is to use a `JacksonAnnotationIntrospector` to ignore the `@JsonIdentityInfo` annotations. I don't think there has been a response from the contributors however, and I am looking for alternative solutions that are a bit more flexible?

Would it be an idea to introduce a `SerializationFeature.USE_OBJECT_ID`? Or perhaps a `@JsonIdentityReference(neverAsId = true)` for deserialisation use-cases that rely on a [`ObjectIdResolver`](https://github.com/vzhyhunou/vzh-cms/blob/master/src/main/java/vzh/cms/model/IdResolver.java).

**Comment by cowtowncoder** [comment]
To me, use of `AnnotationIntrospector` is the way to go, and I am not sold on adding features to disable things that are explicitly marked via annotations. If Object Ids are not to be used, `@JsonIdentityInfo` should not be added.

Another possibility for "optional" use are mix-in annotations: `@JsonIdentityInfo` annotation may be added via mix-in just like any other Jackson annotation; so adding to a base class when needed could work.

**Reply by jwgmeligmeyling** [reply]
Fair, in retrospect I guess I was kind of abusing `JsonIdentityReference` for its deserialisation possibilities with `resolver` and I ended up writing custom Deserializers and a `BeanDeserializerModifier` to get the job done instead 👍🏻
